### PR TITLE
feat(FOROME-802): add a dnd scroll to the drawer and a sticky left column in it

### PR DESCRIPTION
**Pull request by QYaroslavTrefilov**
https://quantori.atlassian.net/jira/software/c/projects/FOROME/boards/202?modal=detail&selectedIssue=FOROME-802&assignee=61387ba9156ab300717f62e9

In this PR i only add dnd scroll for drawer window. Another parts of redesign  will be added at other tasks

### CODE REVIEW

File: src/components/variant/ui/drawer-window.tsx [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+import { IGridLayout } from '@declarations'
+import { t } from '@i18n'
+import {
+  ICommonAspectDescriptor,
+  IPreAspectDescriptor,
+  ITableAspectDescriptor,
+  TRecCntResponse,
+} from '@service-providers/dataset-level/dataset-level.interface'
+
+const normClass = 'norm'
```
Comment (QSdmitrioul) [inline src/components/variant/ui/drawer-window.tsx:18]: Why not enum?

File: src/components/variant/ui/drawer-window.tsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+const TableView = ({
+  colhead,
+  rows,
+  name,
+  shouldAddShadow,
+}: ITableViewProps): ReactElement => {
+  let colheadData: string[] = []
+
+  if (colhead) {
+    colheadData = [colhead?.[0]?.[0]]
```
Comment (QSdmitrioul) [inline src/components/variant/ui/drawer-window.tsx:43]: you can push element without reassigning it, so you can use `const colheadData`

File: src/components/variant/ui/drawer-window.tsx [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+    const handleScroll = () => {
+      const currentLeftDistance = getLeftDistance(ref.current)
+
+      if (!currentLeftDistance) return
+
+      const fixedCurrentLeftDistance = Math.round(currentLeftDistance)
+
+      if (fixedCurrentLeftDistance !== startedLeftDistance) {
+        setShouldAddShadow(true)
```
Comment (QSdmitrioul) [inline src/components/variant/ui/drawer-window.tsx]: you can swap conditions(if elements not bool), add return statement and remove second condition

File: src/components/variant/ui/drawer-window.tsx (hunk 1 above)
Comment (QYaroslavTrefilov) [reply src/components/variant/ui/drawer-window.tsx:18]: This structure existed before me. In order not to increase the PR in size, I decided not to make minor changes

File: src/components/variant/ui/drawer-window.tsx (hunk 2 above)
Comment (QYaroslavTrefilov) [reply src/components/variant/ui/drawer-window.tsx:43]: Same situation. I didnt change this logick in this PR

**Review by QSdmitrioul — APPROVED** [review]

File: src/components/variant/ui/drawer-window.tsx [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+              )
+            })}
+          </tbody>
+        </table>
+      )}
+    </div>
+  )
+}
+
+export const DrawerWindow = observer(
```
Comment (AndreyUstyumenko) [inline src/components/variant/ui/drawer-window.tsx:158]: 3 different components in a single file. Better to split it on to 3

File: src/components/variant/ui/drawer-window.tsx [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+
+    const getLeftDistance = (element: HTMLDivElement | null): number | null => {
+      const tableNode = element?.children?.[0]?.children?.[0]
+      const tbodyNode = tableNode?.children[tableNode?.children.length - 1]
+      const trackedTdNode = tbodyNode?.children?.[0]?.children?.[1]
+
+      if (!trackedTdNode) return null
+
+      return trackedTdNode.getBoundingClientRect().left
+    }
```
Comment (AndreyUstyumenko) [inline src/components/variant/ui/drawer-window.tsx:176]: does not belong to the component, could be a utils function

File: src/components/variant/ui/drawer-window.tsx [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+    >(null)
+
+    const [shouldAddShadow, setShouldAddShadow] = useState(false)
+
+    const getLeftDistance = (element: HTMLDivElement | null): number | null => {
+      const tableNode = element?.children?.[0]?.children?.[0]
+      const tbodyNode = tableNode?.children[tableNode?.children.length - 1]
+      const trackedTdNode = tbodyNode?.children?.[0]?.children?.[1]
+
+      if (!trackedTdNode) return null
```
Comment (AndreyUstyumenko) [inline src/components/variant/ui/drawer-window.tsx:173]: why necessarily return null? You can return `trackedTdNode?.getBoundingClientRect().left` which would be undefined in a negative case

File: src/components/variant/ui/drawer-window.tsx [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+      const currentLeftDistance = getLeftDistance(ref.current)
+
+      if (!currentLeftDistance) return
+
+      const fixedCurrentLeftDistance = Math.round(currentLeftDistance)
+
+      const isStartPosition = fixedCurrentLeftDistance === startedLeftDistance
+
+      setShouldAddShadow(!isStartPosition)
+    }
```
Comment (AndreyUstyumenko) [inline src/components/variant/ui/drawer-window.tsx:198]: maybe move handlers to a custom hook, but it's up to you

**Pull request merged**